### ISSUE TITLE
MS Word unusual lag after enabling UI autimation support in nvda advance setting

### ISSUE THREAD

**Issue by LittleStar-VIP** [nvaccess/nvda#13461]
### Steps to reproduce:
1. Open nvda setting
2. go to advance setting 
3. make sure using UI Autimation to process word controls is checked (sorry, I am not using english version of nvda, I am not sure the exact english name of this item)
4. open word and create a file with more than 20k words with a table containing all these words, like around 700 rows with two columns
5. after pressing tab to jump to next new row, nvda willbe busy for around half a minute before nvda can give anyspeech, at this moment, nvda cannot interact with word
6. when the numberof  rows increase,the lag time may increase
7. sometimes after pressing tab and waitedafter the lag time has past, nvda do report the new row number and column number, but sometimes do not

After a bit of time of try and error, I found that if I disabled the function "using UI Autimation to process word controls" the problem is no longer existed 

### Expected behavior:
pressing tab will immediately jump to next new row without lag

#### NVDA installed/portable/running from source:
installed
#### NVDA version:
nvda 2021.3.3
#### Windows version:
Windows 10 home 21H1
#### Name and version of other software in use when reproducing the issue:
MS Word 365 installed version
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
I have tried creating a clean portable copy of nvda and this problem do not occur

#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Not tried

**Comment by cary-rowen** [comment]
Hi, I did some more testing on this and found the following:
1. The problem can be reproduced using the latest alpha-24937;
2. After creating the portable version, changing "Use UI Automation to access Microsoft Word document controls" to "Always" can still reproduce.
3. When I press Tab to create a new row in the last row of the table, NVDA will freeze for half a minute, and pressing ctrl+z to undo the operation of adding a new row will still freeze for half a minute.
4. Microsoft 365MSO (version 2202 Build 16.0.14931.20116) 64 bit;

Below is my test sample:
[test.zip](https://github.com/nvaccess/nvda/files/8231298/test.zip)

**Comment by cary-rowen** [comment]
cc @michaelDCurran @leonardder

**Comment by Adriani90** [comment]
This is still reproducible with NVDA 2023.1 RC1, I get a lag of 10 seconds even on a computer with 32gb of Ram and Intel 12900HK with 5 ghz, 14 cores and 24 threats.

**Comment by Adriani90** [comment]
The performance lag is not reproducible anymore with UIA on, I am closing as works for me. Tested with NVDA 2024.4.1. If you are still having this issue, please comment and we can reopen.
I tested using tab, ctrl+alt+arrow key, arrow keys or ctrl+arrow keys, none of these navigation paterns caused performance lag in MS Word.
With UIA off though ctrl+alt+arrow keys seems a bit slower, but there is no unusual lag.
